### PR TITLE
[MM-51053] delete helm to cleanup LBs during cluster teardown

### PR DESCRIPTION
**Pull request by mirshahriar**
#### Summary

We need to delete helm to cleanup LBs while deleting cluster.

#### Ticket Link

  Fixes https://mattermost.atlassian.net/browse/MM-51053

#### Release Note

```release-note
None
```

### CODE REVIEW

File: internal/provisioner/nginx.go [hunk 1]
```diff
@@ -116,7 +116,11 @@ func (n *nginx) ActualVersion() *model.HelmUtilityVersion {
 }
 
 func (n *nginx) Destroy() error {
-	return nil
+	helm, err := n.NewHelmDeployment()
+	if err != nil {
+		return err
+	}
+	return helm.Delete()
```
Comment (gabrieljackson) [inline internal/provisioner/nginx.go:123]: I believe the helm client would return immediately after calling delete, but k8s may need some time to clean things up in the background afterwards. Do you know if that is correct or not? I am wondering if we need to add some wait logic to ensure there is enough time for a full cleanup.

File: internal/provisioner/helm_utils.go [hunk 2]
```diff
@@ -94,7 +94,7 @@ func (d *helmDeployment) Exists() (bool, error) {
 	}
 
 	for _, chart := range list.asSlice() {
-		if chart.Name == d.chartDeploymentName && chart.Chart == d.chartName && chart.Namespace == d.namespace {
```
Comment (fmartingr) [inline internal/provisioner/helm_utils.go:97]: What was the problem with this check? just curious.

**Review by fmartingr — APPROVED** [review]
Changes look good to me, just wondering what implies waiting for all charts to get removed now. This mostly affects test/dev on a daily basis, but still.

File: internal/provisioner/helm_utils.go (hunk 2 above)
Comment (mirshahriar) [reply internal/provisioner/helm_utils.go:97]: They are different entities. `d.chartName` holds `ingress-nginx/ingress-nginx` but `chart.Chart` holds `ingress-nginx-4.2.0`

**Comment by mirshahriar** [comment]
> what implies waiting for all charts to get removed now

In current version, It takes 6 minutes to delete a **dev** cluster. If we delete all helms, it takes 10 minutes. However, deleting all of these helms concurrently (**goroutine**) only takes 5 minutes.

@fmartingr do you think we should go for `goroutine`?

**Comment by fmartingr** [comment]
> > what implies waiting for all charts to get removed now
> 
> In current version, It takes 6 minutes to delete a **dev** cluster. If we delete all helms, it takes 10 minutes. However, deleting all of these helms concurrently (**goroutine**) only takes 5 minutes.
> 
> @fmartingr do you think we should go for `goroutine`?

With the numbers you provided, I don't think is worth it to be honest.

**Review by gabrieljackson — APPROVED** [review]
LGTM

**Pull request merged**